cdc/owner.go

```diff
@@ -360,15 +373,36 @@ func (o *ownerImpl) addCapture(info *model.CaptureInfo) {
 }
 
 func (o *ownerImpl) handleMarkdownProcessor(ctx context.Context) {
-	for id := range o.markDownProcessor {
-		err := DeleteCaptureInfo(ctx, id, o.etcdClient)
+	var deletedCapture = make(map[string]struct{})
+	for i := len(o.markDownProcessor) - 1; i >= 0; i-- {
+		snap := o.markDownProcessor[i]
+		changefeed, ok := o.changeFeeds[snap.cfID]
+		if !ok {
+			log.Error("changefeed not found in owner cache, can't rebalance",
+				zap.String("changefeedID", snap.cfID))
+			continue
+		}
+		for _, tbl := range snap.tables {
+			changefeed.reAddTable(tbl.ID, tbl.StartTs)
+		}
+		err := kv.DeleteSubChangeFeedInfo(ctx, o.etcdClient, snap.cfID, snap.captureID)
 		if err != nil {
-			log.Warn("failed to delete key", zap.Error(err))
+			log.Warn("failed to delete subchangefeed info",
+				zap.String("changefeedID", snap.cfID),
+				zap.String("captureID", snap.captureID),
+				zap.Error(err),
+			)
 			continue
 		}
+		o.markDownProcessor = append(o.markDownProcessor[:i], o.markDownProcessor[i+1:]...)
```

Comment:
It seems like most of the time all `markDownProcessor` should be removed, so it might be simpler to keep track of what items remain after all items are checked and then assign it to `o.markDownProcessor`.